src/styles/c-global-style.scss

```diff
@@ -90,19 +90,34 @@ $interaction-types3: primary, dark;
     }
   }
 }
-.bg {
-  @each $type in join(join($interaction-types, $interaction-types2), $interaction-types3) {
-    &-#{$type} {
-      background-color: map-get($interaction-colors, $type) !important;
-      background-color: var(--#{$type}) !important;
+
+$util-elems : bg, text, border;
+
+@each $elem in $util-elems {
+  .#{$elem} {
+    @if($elem == bg) {$elem : background;}
+    @each $type in join(join($interaction-types, $interaction-types2), $interaction-types3) {

```

Comment:
I think it would make more sense to move the logic here to the already existing loop.
Then run the other loop inside.

Reply:
We will separate buttons selector and its logic because they contain more interaction pattern such as on hover, on disabled, on active, etc.